crates/oxc_cli/src/lib.rs

```diff
@@ -59,33 +62,41 @@ impl Cli {
             })
             .collect::<Vec<_>>();
 
-        let number_of_warnings = paths
-            .par_iter()
-            .map(|path| {
-                let diagnostics = Self::lint_path(path);
-                diagnostics
-                    .iter()
-                    .filter(|d| match d.severity() {
-                        // The --quiet flag follows ESLint's --quiet behavior as documented here: https://eslint.org/docs/latest/use/command-line-interface#--quiet
-                        // Note that it does not disable ALL diagnostics, only Warning diagnostics
-                        Some(Severity::Warning) => !self.cli_options.quiet,
-                        _ => true,
-                    })
-                    .for_each(|diagnostic| {
-                        println!("{diagnostic:?}");
-                    });
+        let (_, (warnings, diagnostics)): (_, (usize, usize)) = rayon::join(
+            move || {
+                paths.par_iter().for_each(|path| {
+                    let diagnostics = Self::lint_path(path);
 
-                diagnostics.len()
-            })
-            .sum();
+                    for d in diagnostics {
+                        sender.send(d).unwrap();
+                    }
+                });
+            },
+            move || {
+                let mut number_of_warnings = 0;
+                let mut number_of_diagnostics = 0;
+
+                while let Ok(diagnostic) = receiver.recv() {
+                    // The --quiet flag follows ESLint's --quiet behavior as documented here: https://eslint.org/docs/latest/use/command-line-interface#--quiet
+                    // Note that it does not disable ALL diagnostics, only Warning diagnostics
+                    if diagnostic.severity() == Some(Severity::Warning) && !self.cli_options.quiet {
+                        number_of_warnings += 1;
+                    }
+                    number_of_diagnostics += 1;
+                    println!("{diagnostic:?}");
```

Comment:
are we missing the logic where it stops printing when `warnings > max_warnings` ?

Reply:
Is the `--quiet` logic also absent?

Reply:
Oops, should be fixed now